unittests/test_environments.py

```diff
@@ -88,6 +88,15 @@ def test_load_restore(self):
 
         assert not self.environ.is_loaded
 
+    @fixtures.switch_to_user_runtime
+    def test_temp_environment_context_manager(self):
+        with env.temp_environment(variables=[('_var0', 'val2'),
+                                             ('_var3', 'val3')],
+                                  modules=['testmod_foo']):
+            assert os.environ['_var0'] == 'val2'
+            assert os.environ['_var3'] == 'val3'
```

Comment:
It'd be easier to check that the environment is set inside the context manager, if `__enter__` returned the actual environment that was created, which btw makes more sense to be returned instead of the snapshot. Then you could do the following:

```python
with env.temp_environment(...) as environ:
    assert environ.is_loaded

assert not environ.is_loaded
```